components/ai_chat/content/browser/page_content_fetcher.cc

```diff
@@ -104,6 +127,33 @@ class PageContentFetcher {
         std::move(callback), invalidation_token));
   }
 
+  void StartGithub(
+      GURL patch_url,
+      scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory,
+      FetchPageContentCallback callback) {
+    url_loader_factory_ = url_loader_factory;
+    auto request = std::make_unique<network::ResourceRequest>();
+    request->url = patch_url;
+    request->load_flags = net::LOAD_DO_NOT_SAVE_COOKIES;
+    request->credentials_mode = network::mojom::CredentialsMode::kInclude;
+    request->request_initiator = absl::nullopt;
```

Comment:
this was done because of a network trace for a top level load of .patch, it matches the value for that when I set it like this.

Reply:
I can probably remove that line. I’ll check.  I didn’t know why at first when I was developing it, but a top level load would get a private repo patch but my code was not. So I did a network-log to see what the differences were. And I set things so it appeared as a top level load. I think this is the only bit that actually matters for that `request->credentials_mode = network::mojom::CredentialsMode::kInclude;`

Reply:
``` 
  // Failure to provide a correct, verified |request_initiator| may
  // lead to bypasses of CORS, ORB, SameSite cookies and other HTTP security
  // features.  An untrustworthy process (e.g. a renderer or a utility process)
  // should not be able to trigger or influence requests with a absl::nullopt
  // |request_initiator|).
  ```
  
from https://source.chromium.org/chromium/chromium/src/+/main:services/network/public/mojom/url_request.mojom;l=161-169?q=url_request.mojom&ss=chromium it seems most appropriate to set the request_initiator to the PR URL. 

it seems `absl::nullopt` should really be used only for user-initiated requests like direct navigation in the omnibox or clicking a bookmark.

i don't see any security issues in the way it is now, so this is non-blocking.

Reply:
I originally used `absl::nullopt` when developing the feature because it was retrieving the patch file via a top level load and not from my code. And I noticed not setting it is different from setting it to `absl::nullopt` in the network log, so I was trying to make it appear as top level.  But that wasn't needed. I'll leave it as is though with the line removed now because our other usages do the same.